java/test/src/main/java/io/ray/docdemo/WalkthroghDemo.java

```diff
@@ -0,0 +1,148 @@
+package io.ray.docdemo;
+
+import com.google.common.collect.ImmutableList;
+import io.ray.api.ActorHandle;
+import io.ray.api.ObjectRef;
+import io.ray.api.Ray;
+import io.ray.api.WaitResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.testng.Assert;
+
+/**
+ * This class contains demo code of the Ray core walkthrough doc (https://docs.ray.io/en/latest/walkthrough.html).
+ *
+ * Please keep them in sync.
+ */
+public class WalkthroghDemo {
+
+  public static class MyRayApp {
+
+    // A regular Java static method.
+    public static int myFunction() {
+      return 1;
+    }
+
+    public static int slowFunction() throws InterruptedException {
+      TimeUnit.SECONDS.sleep(10);
+      return 1;
+    }
+
+    public static int functionWithAnArgument(int value) {
+      return value + 1;
+    }
+  }
+
+  public static void demoTasks() {
+    // Invoke the above method as a Ray remote function.
+    // This will immediately returns an object ref (a future) and then creates
+    // a task that will be executed on a worker process.
+    ObjectRef<Integer> res = Ray.task(MyRayApp::myFunction).remote();
+
+    // The result can be retrieved with ``ray.get``.
+    Assert.assertTrue(res.get() == 1);
+
+    // Invocation of Ray remote functions happen in parallel.
+    // All computation is performed in the background, driven by Ray's internal event loop.
+    for (int i = 0; i < 4; i++) {
+      // This doesn't block.
+      Ray.task(MyRayApp::slowFunction).remote();
+    }
+
+    ObjectRef<Integer> objRef1 = Ray.task(MyRayApp::myFunction).remote();
+    Assert.assertTrue(objRef1.get() == 1);
+
+    // You can pass an `ObjectRef` as an argument to another Ray remote function.
+    ObjectRef<Integer> objRef2 = Ray.task(MyRayApp::functionWithAnArgument, objRef1).remote();
+    Assert.assertTrue(objRef2.get() == 2);
```

Comment:
```suggestion
    Assert.assertEquals(objRef2.get(), 2);
```

Reply:
I used this originally, but `Assert.assertEquals` doesn't support comparing `int` and `Integer`.